File: Arrays/P003.py
def brute(arr):
    n=len(arr)
    for i in range(n):
        for j in range(i+1,n):
            if arr[i]>arr[j]:
                return False
    return True

def optimal(arr):
    n=len(arr)
    for i in range(n-1):
        if arr[i]>arr[i+1]:
            return False
    return True

if __name__=='__main__':
    arr=input('Enter elements: ')
    arr=arr.split()
    print('Brute: ',brute(arr))
    print('Optimal: ',optimal(arr))

File: Stiver_Problems/Arrays/P003.cpp
// P003.cpp - Check if array is sorted in non-decreasing order

#include <bits/stdc++.h>
using namespace std;

bool brute(const vector<int>& arr) {
    int n = arr.size();
    for (int i = 0; i < n; i++) {
        for (int j = i + 1; j < n; j++) {
            if (arr[i] > arr[j]) return false;
        }
    }
    return true;
}

bool optimal(const vector<int>& arr) {
    int n = arr.size();
    for (int i = 0; i < n - 1; i++) {
        if (arr[i] > arr[i + 1]) return false;
    }
    return true;
}

int main() {
    cout << "Enter elements: ";
    string line;
    getline(cin, line);
    stringstream ss(line);
    vector<int> arr;
    int x;
    while (ss >> x) arr.push_back(x);

    cout << "Brute: " << (brute(arr) ? "True" : "False") << "\n";
    cout << "Optimal: " << (optimal(arr) ? "True" : "False") << "\n";
    return 0;
}
